clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -44,7 +44,11 @@
      * @return JAAS configuration object
      */
     public static String jaasConfig(String loginContextName, String key) throws IOException {
```

Comment:
Moved the loading of `Configuration` so that `Configuration.getConfiguration` is only called from `JaasUtils` now. Since the KIP (and hence the PR) doesn't change configuration loading on the server side, the singleton still remains. When the extensions in KIP-103 to support multiple configs on the server are added, we may be able to clean it up some more.